browser-extension/mv2/src/background/background.js

```diff
@@ -1143,6 +1143,37 @@ BG.Methods.addListenerForExtensionMessages = function () {
 };
 
 BG.Methods.handleClientPortConnections = () => {
+  chrome.webNavigation.onCommitted.addListener((navigatedTabData) => {
+    const tabId = navigatedTabData.tabId;
+    if (!tabId) {
+      return;
+    }
+
+    const clientPortData = window.tabService.getData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT);
+    if (
+      clientPortData?.sender?.documentLifecycle !== "active" &&
+      navigatedTabData.frameId === 0 &&
+      navigatedTabData.documentLifecycle === "active"
+    ) {
+      window.tabService.setData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT, {
+        ...clientPortData,
+        sender: {
+          ...clientPortData.sender,
+          documentLifecycle: navigatedTabData.documentLifecycle,
+        },
+      });
+
+      const clientLoadSubscribers = window.tabService.getData(tabId, BG.TAB_SERVICE_DATA.CLIENT_LOAD_SUBSCRIBERS) || [];
+      window.tabService.removeData(tabId, BG.TAB_SERVICE_DATA.CLIENT_LOAD_SUBSCRIBERS);
+      clientLoadSubscribers.forEach((subscriber) => subscriber());
+
+      BG.Methods.onClientPageLoad({
```

Comment:
There was a handling for pre-rendered pages from BF cache in content script (using `pageshow` event). Can we check if that logic is still valid with webNavigated commit hooks?

Test case:
1. Setup some rules on `example.com`. 
2. Open new tab and type `example.com`. Notice, there is no change in extension icon before hitting enter.
3. After page loads, extension icon should turn green.
4. Click browser back and forth.
5. Click More info link in example.com and click browser back and forth.

The extension icon and Executed Rules tab in popup should show correct information.